torch/profiler/profiler.py

```diff
@@ -44,6 +46,20 @@ def supported_activities():
     return torch.autograd._supported_activities()
 
 
+def _optimizer_post_hook(optimizer, args, kwargs):
+    KinetoStepTracker.increment_step("Optimizer")
+
+
+def _profile_using_dynolog(override: bool = False):
```

Comment:
Do you need to guard this to run only once? E.g. something like
```
_dynolog_handle = None

def _profile_using_dynolog(override: bool = False):
    global _dynolog_handle
    if _dynolog_handle is not None and (...):
        _dynolog_handle = ...
    
```

Reply:
this makes sense. done.